compiler/src/iree/compiler/Dialect/Util/Transforms/HoistIntoGlobals.cpp

```diff
@@ -29,6 +30,47 @@ static llvm::cl::opt<std::string> clPrintDotGraphToFile(
         "nodes represent roots and the green nodes represent hoisted values."),
     llvm::cl::value_desc("filename"));
 
+template <typename AccessorTy>
+static inline bool isAccessorParameterized(SymbolTable moduleSymbols,
+                                           AccessorTy op) {
+  auto global = moduleSymbols.lookup(op.getGlobalName());
+  if (!global) {
+    return true;
+  }
+  auto utilGlobal = cast<Util::GlobalOpInterface>(global);
+  if (!utilGlobal.getGlobalInitialValue()) {
+    return false;
+  }
+  return !isa<Util::SerializableAttrInterface>(
+      utilGlobal.getGlobalInitialValue());
+}
+
+// Today the only way to interact with a global is with loads, stores, and
+// addresses, and globals are the only way to reference parameters given where
+// const-eval is run today. This is a workaround until we have proper dialect
+// interfaces for detecting whether something is evaluatable at compile time.
+static bool isParameterized(SymbolTable moduleSymbols, Operation *initializer) {
+  WalkResult res = initializer->walk([&](Operation *op) {
```

Comment:
this won't handle calls that load parameters - I think DFX is needed here unless calls prevent hoisting (which ideally they wouldn't, but if this isn't handling calls properly today we should just make sure it doesn't incorrectly try)

Reply:
Short answer is, I'm not sure hoisting handles calls properly period, so easiest thing to do here is probably to bail on calls. Probably need to fix the analysis to not hoist function calls for the time being as well.

The correct solution here should let the analysis tell us whether an expression tree is compile time evaluatable so that we don't have to redo the walk here (I think, maybe we still end up needing the walk/explore).

Reply:
I just bailed on calls. Need to fix the analysis first to handle calls properly, but this at least fixes the bug for disabling const-eval with parameters.